### PR TITLE
Add test for IPC

### PR DESCRIPTION
**Pull request by ArdeleanuStefan**

### CODE REVIEW

**Review by maxlig79 — APPROVED** [review]

File: ArdeleanuStefan_Task3/src/test.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+std::ifstream client1{"tmp/log.0"};
+std::ifstream client2{"tmp/log.1"};
+
+BOOST_AUTO_TEST_SUITE(ClientServerIPCTest)
+
+BOOST_AUTO_TEST_CASE(test_1)
+{
+    system("./src/script.sh ");
+    system("echo 'add;unu\nget;0\ndelete;doi' > tmp/input.pipe");
+    system("echo 'add;doi\nget;1\ndelete;doi' > tmp/input1.pipe");
```
Comment (maxlig79) [inline ArdeleanuStefan_Task3/src/test.cpp]: I would like to send both command lists simultaneously, in order to check synchronization.

File: ArdeleanuStefan_Task3/src/test.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    std::string output;
+    int count = 0;
+    while (std::getline(client1, output))
+    {
+        switch (count)
+        {
+        case 0:
+            BOOST_CHECK(output == "Add OK");
```
Comment (maxlig79) [inline ArdeleanuStefan_Task3/src/test.cpp:21]: If the client1 file is empty or has only one of the expected checks the test will pass.

File: ArdeleanuStefan_Task3/src/test.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        count++;
+    }
+
+    count = 0;
+    while (std::getline(client2, output)){
+        switch (count)
+        {
+        case 0:
+            BOOST_CHECK(output == "Add OK");
```
Comment (maxlig79) [inline ArdeleanuStefan_Task3/src/test.cpp:43]: If the client2 file is empty or has only one of the expected checks the test will pass.

File: ArdeleanuStefan_Task3/src/test.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            break;
+        default:
+            break;
+        }
+
+        count++;
+    }
+
+    system("echo 'exit' > tmp/input.pipe");
+    system("echo 'exit' > tmp/input1.pipe");
```
Comment (maxlig79) [inline ArdeleanuStefan_Task3/src/test.cpp]: After running the test booth server and client are still running.

**Review by maxlig79 — DISMISSED** [review]

**Review by silvanadomnica — DISMISSED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by silvanadomnica — APPROVED** [review]

**Pull request merged**